src/cdk-experimental/ui-patterns/behaviors/list-selection/list-selection.ts

```diff
@@ -0,0 +1,142 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.dev/license
+ */
+
+import {signal, Signal, WritableSignal} from '@angular/core';
+import {ListNavigation, ListNavigationItem} from '../list-navigation/list-navigation';
+
+/** The required properties for selection items. */
+export interface ListSelectionItem extends ListNavigationItem {
+  /** A unique identifier for the item. */
+  id: Signal<string>;
+
+  /** Whether an item is disabled. */
+  disabled: Signal<boolean>;
+}
+
+/** The required inputs for list selection. */
+export interface ListSelectionInputs<T extends ListSelectionItem> {
+  /** The items in the list. */
+  items: Signal<T[]>;
+
+  /** Whether multiple items in the list can be selected at once. */
+  multiselectable: Signal<boolean>;
+
+  /** The ids of the current selected items. */
+  selectedIds: WritableSignal<string[]>;
+
+  /** The selection strategy used by the list. */
+  selectionMode: Signal<'follow' | 'explicit'>;
+}
+
+/** Controls selection for a list of items. */
+export class ListSelection<T extends ListSelectionItem> {
+  /** The id of the previous selected item. */
+  anchorId = signal<string | null>(null);
```

Comment:
```suggestion
  previouslySelectedId = signal<string | null>(null);
```
name seems out of sync with the comment?

Reply:
Would "lastSelectedId" be ok here?

Reply:
That's probably fine; as a matter of taste I prefer using "previous" or "mostRecent" instead of "last" for names like this because "last" could also mean e.g. "last item in a list"